ui/src/dashboards/components/QueryEditor.tsx

```diff
@@ -0,0 +1,465 @@
+import React, {Component, MouseEvent} from 'react'
+import _ from 'lodash'
+import classnames from 'classnames'
+
+import TemplateDrawer from 'src/shared/components/TemplateDrawer'
+import QueryStatus from 'src/shared/components/QueryStatus'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import {Controlled as ReactCodeMirror, IInstance} from 'react-codemirror2'
```

Comment:
would you consider organizing these types like in `DashboardPage`, i.e.:
```
// Libraries

// Components

// Actions

// Utils

// APIs

// Constants

// Types
```

-- would make it easier to know what to expect throughout this file